server/api/classyear/classyear.controller.js

```diff
@@ -220,5 +222,20 @@ function generateCode(codeLength){
       var character = (Math.floor(Math.random() * characterOptions.length));
       code = code.concat(characterOptions[character.toString()]);
   }
+  ClassYear.findOne({"dayCodes.code":code})
```

Comment:
@agundy if a duplicate code was found, it would call itself to get another code.